packages/mock-block-dock/src/debug-view/dev-tools/datastore-graph-visualisation.tsx

```diff
@@ -0,0 +1,196 @@
+import { Entity, Link } from "@blockprotocol/graph/.";
+import { GraphChart, GraphSeriesOption } from "echarts/charts";
+import * as echarts from "echarts/core";
+import { SVGRenderer } from "echarts/renderers";
+import { useEffect, useRef, useState } from "react";
+
+import { useMockBlockDockContext } from "../../mock-block-dock-context";
+
+const parseNameFromEntity = (entity: Entity): string =>
+  typeof entity.properties.name === "string"
+    ? entity.properties.name
+    : entity.entityId;
+
+const graphHeight = 500;
+
+type SeriesOption = GraphSeriesOption;
+
+// Combine an Option type with only required components and charts via ComposeOption
+type EChartOption = echarts.ComposeOption<SeriesOption>;
+
+const createDefaultEChartOptions = (params?: {
+  initialNodes?: GraphSeriesOption["nodes"];
+  initialEdges?: GraphSeriesOption["edges"];
+}): EChartOption => ({
+  color: ["#6F59EB"],
+  series: [
+    {
+      type: "graph",
+      layout: "force",
+      draggable: true,
+      /** @todo: find way to get this working, currently interferes with `draggable` */
+      // roam: true,
```

Comment:
When setting `draggable: true` (so that nodes can be dragged around) and `roam: true` (so that the user can pan around the visualisation using their mouse) buggy behaviour occurs in the ECharts visualisation:

https://user-images.githubusercontent.com/42802102/190125968-8eaae7aa-432c-4135-8b3c-6cef5d9da78b.mov

What I've opted to do instead is to not set `roam: true` for now,